httpx/client.py

```diff
@@ -132,9 +132,12 @@ def merge_cookies(
             auth = HTTPBasicAuth(username=url.username, password=url.password)
 
         if auth is not None:
-            if isinstance(auth, tuple):
-                auth = HTTPBasicAuth(username=auth[0], password=auth[1])
-            request = auth(request)
+            if isinstance(auth, HTTPDigestAuth):
+                self.auth = auth
+            else:
+                if isinstance(auth, tuple):
+                    auth = HTTPBasicAuth(username=auth[0], password=auth[1])
+                request = auth(request)
```

Comment:
Are we ok with allowing the users to pass instances of `HTTPBasicAuth` and `HTTPDigestAuth`?

Reply:
Yes.